internal/pubsub/acks_test.go

```diff
@@ -266,3 +263,26 @@ func TestConcurrentDoubleAckCausesPanic(t *testing.T) {
 		t.Errorf("panics = %d, want %d", panics, 1)
 	}
 }
+
+func TestSubShutdownCanBeCanceledEvenWithHangingSendAcks(t *testing.T) {
```

Comment:
This test should verify that canceling the Shutdown context results in:
- outstanding SendAcks calls (of which there must be one) returning quickly
- Shutdown itself returning quickly

So there should be a `cancel` call somewhere, with Shutdown running in a goroutine and then closing or sending to a channel, and lastly something like
```
select {
    case <-shutdownDone:
    case <-time.After(5*time.Second):
           t.Fatal(...)
}
```
Yes, it's time sensitive, but the idea is to avoid the entire test running for 10 minutes and then timing out.

Reply:
Done.